content/guides/nextjs/github-open-authoring/configure-custom-app.md

```diff
@@ -0,0 +1,118 @@
+     */
+    this.cms = new TinaCMS({
+      apis: {
+        /**
+         * 2. Register the GithubClient
+         */
+        github: new GithubClient({
+          proxy: '/api/proxy-github',
+          authCallbackRoute: '/api/create-github-access-token',
+          clientId: process.env.GITHUB_CLIENT_ID,
+          baseRepoFullName: process.env.REPO_FULL_NAME, // e.g: tinacms/tinacms.org,
+        }),
+      },
+      /**
+       * 3. Make sure the Sidebar & Toolbar are
+       *    hidden unless we're in Preview/Edit Mode
+       */
+      sidebar: {
+        hidden: !props.pageProps.preview,
+      },
+      toolbar: {
+        hidden: !props.pageProps.preview,
+      },
+    })
+  }
+
+  render() {
+    const { Component, pageProps } = this.props
+    return (
+      /**
+       * 4. Wrap the page Component with the Tina and Github providers
+       */
+      <TinaProvider cms={this.cms}>
+        <TinacmsGithubProvider
+          editMode={pageProps.preview}
+          enterEditMode={enterEditMode}
+          exitEditMode={exitEditMode}
+          error={pageProps.error}
+        >
+          {/**
+           * 5. Add a button for entering Preview/Edit Mode
+           */}
+          <EditLink editMode={pageProps.preview} />
+          <Component {...pageProps} />
+        </TinacmsGithubProvider>
+      </TinaProvider>
+    )
+  }
+}
+
+const enterEditMode = () => {
+  return fetch(`/api/preview`).then(() => {
+    window.location.href = window.location.pathname
+  })
+}
+
+const exitEditMode = () => {
+  return fetch(`/api/reset-preview`).then(() => {
+    window.location.reload()
+  })
+}
+
+export interface EditLinkProps {
+  editMode: boolean
+}
+
+export const EditLink = ({ editMode }: EditLinkProps) => {
+  const github = useGithubEditing()
+
+  return (
+    <button onClick={editMode ? github.exitEditMode : github.enterEditMode}>
+      {editMode ? 'Exit Edit Mode' : 'Edit This Site'}
+    </button>
+  )
+}
+```
+
+> _Note:_ For brevity, the example above configures many steps in a single file, but **a few components can be configured in different places**. For example you could put the `EditLink` in a Layout component, or set up the Github Provider only on certain pages.
+
+If you restart the dev server, you should see a **button in the top left-hand corner** that says, "Edit This Site". If you click it, you'll be _prompted to authenticate_ with GitHub. If auth is successful, you should see another modal prompting you to create a fork, go ahead and do so.
+
+![github-create-fork-step](/img/github-open-auth-cna/create-fork-step.png)
+
+Since you're the owner of this repository, your edits will go to the `master` branch. If you weren't the owner of this repository, a fork (also known as a _Working Repository_) would be made where someone else could commit edits and create a PR for review.
+
+After you've authenticated and the _Working Repository_ is established, your `create-next-app` should look something like the image below:
```

Comment:
This isn't accurate anymore. The toolbar doesn't show up until we modify the home page to send the `preview` variable

Reply:
Okay, I guess I should run through it again to see how it's working

Reply:
Man, I wish there was a way to give feedback at this step that that the _Working Repository_ has been set properly.

Reply:
Something like an alert stating 'Your Working Repository has been successfully established'....